pkg/policy/rule.go

```diff
@@ -41,14 +41,16 @@ func (r *rule) validate() error {
 	return nil
 }
 
-func mergeL4Port(ctx *SearchContext, r api.PortRule, p api.PortProtocol, dir string, proto string, resMap L4PolicyMap) int {
+func mergeL4Port(fromEndpoints []api.EndpointSelector, r api.PortRule, p api.PortProtocol,
+	dir string, proto string, resMap L4PolicyMap) int {
+
 	fmt := p.Port + "/" + proto
 	v, ok := resMap[fmt]
 	if !ok {
-		resMap[fmt] = CreateL4Filter(r, p, dir, proto)
+		resMap[fmt] = CreateL4Filter(fromEndpoints, r, p, dir, proto)
 		return 1
 	}
-	l4Filter := CreateL4Filter(r, p, dir, proto)
+	l4Filter := CreateL4Filter(fromEndpoints, r, p, dir, proto)
```

Comment:
> L7 rule will potentially need to be stored per fromEndpoints as the rules should not apply to other endpoints. So while we want to potentially skip/clear fromEndpoints restrictions on pure L4 allow/disallow. We always need to store the L7 rules per fromEndpoints. We will need to encode logic in the proxy which allows matching on the source identity.

@tgraf just to make sure I understand this correctly. If the rule has a number of rules with explicit FromEps lists, we should make sure L7 rules are not merged. However, do we also clear all existing FromEps->L7 entries if there is a rule without explicit set of FromEps?